src/specialized.jl

```diff
@@ -120,19 +120,32 @@ function parse_string(ps::MemoryParserState, b::IOBuffer)
     b
 end
 
-function parse_number(pc::ParserContext, ps::MemoryParserState)
+function parse_number(pc::ParserContext{<:Any,<:Any,AllowNanInf}, ps::MemoryParserState) where AllowNanInf
     s = p = ps.s
     e = length(ps)
     isint = true
+    negative = false
+
+    @inbounds c = ps[p]
+
+    # Parse and keep track of initial minus sign (for parsing -Infinity)
+    if AllowNanInf && c == MINUS_SIGN
+        negative = true
+        p += 1
+    end
 
     # Determine the end of the floating point by skipping past ASCII values
-    # 0-9, +, -, e, E, and .
```

Comment:
why is this removed in comment? - can occur after the initial position, e.g. in 2e-4

Reply:
Fixed in latest push.  (I had remembered this, but forgot to revert this change.)